lib/dispatcher/client.js

```diff
@@ -68,6 +68,10 @@ const getDefaultNodeMaxHeaderSize = http &&
 const noop = () => {}
 
 function getPipelining (client) {
+  if (client[kHTTPContext]?.version === 'h2') {
+    return client[kMaxConcurrentStreams] ?? client[kHTTPContext]?.defaultPipelining ?? 100
+  }
```

Comment:
What remains unclear to me is under which circumstances `getPipelining` can be called with a client that has no `kHTTPContext` present.

In any case, this tries to figure out whether we're using H2 and short circuit on that decision. 

The statement below acts as a fallback but I'm almost 100% sure that `client[kPipelining]` always wins (is never nullish) since a client is always created with on in its constructor. However, the setter for pipelining COULD make it nullish (for some reason) and only then it would fall to the context's setting.